src/test/java/ru/hogwarts/school/StudentControllerTest.java

```diff
@@ -0,0 +1,106 @@
+        Student testStudent = new Student();
+        testStudent.setId(22);
+        testStudent.setAge(22);
+        testStudent.setName("Name");
+        Student oneStudent =  this.testRestTemplate.postForObject("http://localhost:" + port + "/student", testStudent, Student.class);
+        Assertions.assertThat(oneStudent).isNotNull();
+
+    }
+    @Test
+    public void testPostStudent() throws Exception{
+        Student testStudent = new Student();
+        testStudent.setId(22);
+        testStudent.setAge(22);
+        testStudent.setName("Name");
+          Student oneStudent =  this.testRestTemplate.postForObject("http://localhost:" + port + "/student", testStudent, Student.class);
+          Assertions.assertThat(oneStudent).isNotNull();
+
+       Student checkStudent = this.testRestTemplate.getForObject("http://localhost:" + port + "/student/" + oneStudent.getId(), Student.class);
+       Assertions.assertThat(oneStudent.getName()).isEqualTo(checkStudent.getName());
+    }
+    @Test
+    public void testFindStudent() throws Exception{
+        Student testStudent = new Student();
+        testStudent.setId(22);
+        testStudent.setAge(22);
+        testStudent.setName("Name");
+        Student oneStudent = this.testRestTemplate.postForObject("http://localhost:" + port + "/student",testStudent, Student.class);
+        Student checkStudent = this.testRestTemplate.getForObject("http://localhost:" + port + "/student/" + oneStudent.getId(), Student.class);
+        Assertions.assertThat(oneStudent.getId()).isEqualTo(checkStudent.getId());
+
+
+    }
+    @Test
+    public void testDeleteStudent() throws Exception{
+        Student testStudent = new Student();
+        testStudent.setId(22);
+        testStudent.setAge(22);
+        testStudent.setName("Name");
+        Student newStudent = this.testRestTemplate.postForObject("http://localhost:" + port + "/student", testStudent, Student.class);
+        this.testRestTemplate.delete("http://localhost:" + port + "/student/" + newStudent.getId());
+        ResponseEntity<Student> checkError = this.testRestTemplate.exchange("http://localhost:" + port + "/student/" + newStudent.getId(), HttpMethod.GET, null, Student.class);
+        Assertions
+                .assertThat(checkError.getStatusCode()).isEqualTo(HttpStatus.NOT_FOUND);
+    }
+    @Test
+    public void testFindFaculty()throws Exception{
+        Student[] students = testRestTemplate.getForObject("http://localhost:" + port + "/student", Student[].class);
+        Faculty faculty = new Faculty();
+
+        for (Student student:students){
+            Long id = student.getId();
+            faculty = testRestTemplate.getForObject("http://localhost:" + port + "/student/" + student.getId() + "/faculty", Faculty.class);
+            if (faculty.getId() != null) break;
+        }
+
+        assertNotNull(faculty.getId());
+        assertNotNull(faculty.getColor());
+        assertNotNull(faculty.getName());
+    }
```

Comment:
Тест нужно полностью переписать:
```suggestion
    @Test
    public void testFindFaculty() throws Exception {
        ResponseEntity<Faculty> facultyResponseEntity = testRestTemplate
                .postForEntity("/faculty", new Faculty(0L, "faculty", "color"), Faculty.class);
        assertEquals(HttpStatus.OK, facultyResponseEntity.getStatusCode());
        Student request = new Student(0L, "student", 19);
        request.setFaculty(facultyResponseEntity.getBody());
        ResponseEntity<Student> studentResponseEntity = testRestTemplate
                .postForEntity("/student", request, Student.class);
        assertEquals(HttpStatus.OK, studentResponseEntity.getStatusCode());

        ResponseEntity<Faculty> response = testRestTemplate
                .getForEntity("/student/" + studentResponseEntity.getBody().getId() + "/faculty", Faculty.class);
        assertEquals(HttpStatus.OK, response.getStatusCode());
        assertEquals("faculty", response.getBody().getName());
        assertEquals("color", response.getBody().getColor());

    }
```